### PR TITLE
mitmproxy: 0.17.1 -> 1.0.2

### PR DESCRIPTION
**Pull request by fpletz**
###### Motivation for this change

mitmproxy was marked as broken and needed fixing and a version bump.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/mitmproxy/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  };
+
+  propagatedBuildInputs = with python3Packages; [
+    pyopenssl pyasn1 urwid pillow flask click pyperclip blinker
+    construct pyparsing html2text tornado brotlipy requests2
+    sortedcontainers passlib cssutils h2 ruamel_yaml jsbeautifier
+    watchdog editorconfig
+  ];
+
+  doCheck = false;
```
Comment (FRidh) [inline pkgs/tools/networking/mitmproxy/default.nix]: no tests? Please always include a comment with the reason why tests are disabled.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -4122,15 +4122,20 @@ in {
...
+    src = pkgs.fetchFromGitHub {
+      owner = "construct";
+      repo = "construct";
+      rev = "v${version}";
+      sha256 = "1xfmmc5pihn3ql9f7blrciy06y2bwczqvkbcpvh96dmgqwc3wys3";
     };
 
     propagatedBuildInputs = with self; [ six ];
 
+    doCheck = !isPy3k;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: can you include a reason why

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -14627,31 +14632,89 @@ in {
...
+  sortedcontainers = buildPythonPackage rec {
+    name = "sortedcontainers-${version}";
+    version = "1.5.7";
+
+    src = pkgs.fetchurl {
+      url = "mirror://pypi/s/sortedcontainers/${name}.tar.gz";
+      sha256 = "1sjh8lccbmvwna91mlhl5m3z4320p07h063b8x8br4p4cll49w0g";
+    };
 
     doCheck = false;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: can you include the reason

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -22908,16 +22964,34 @@ in {
...
+    version = "0.13.7";
 
     src = pkgs.fetchurl {
       url = "mirror://pypi/r/ruamel.yaml/${name}.tar.gz";
-      sha256 = "0r9mn5lm9dcxpy0wpn18cp7i5hkvjvknv3dxg8d9ca6km39m4asn";
+      sha256 = "1vca2552k0kmhr9msg1bbfdvp3p9im17x1a6npaw221vlgg15z7h";
     };
 
-    propagatedBuildInputs = with self; [ ruamel_base ruamel_ordereddict ];
+    doCheck = false;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: reason

**Review by FRidh — CHANGES_REQUESTED** [review]
That's a lot of updates, thanks. Please fix tests or at least include the reason as a comment for disabling them. I suppose the test suites are run for all other packages with all main interpreter versions?

**Comment by fpletz** [comment]
Added comments with explanations for disabling the tests. I don't have time to fix the tests and packaging new and bumping existing python packages was tedious enough for me.

As indicated above, I have not run `nox-review`, so no idea yet if the bumps break other packages. I want to wait for travis to maybe finish successfully.

**Comment by fpletz** [comment]
Okay, so all reverse dependencies build for me with `nox-review pr` that were not broken before, so we're ready to merge. Also rebased against current master.

**Pull request merged**